src/ArrayListForTaskStorage.java

```diff
@@ -0,0 +1,121 @@
+import java.util.Arrays;
+import java.util.Objects;
+
+class ArrayListForTaskStorage<T> implements ListForTask<T> {
+    private static final int START_SIZE = 10;
+    private T[] values;
+
+    ArrayListForTaskStorage() {
+        this.values = (T[]) new Objects[START_SIZE];
+    }
+
+    private T removeArrayElement(int index) {
+        try {
+            T[] tempArrayList = (T[]) new Object[values.length - 1];
+            System.arraycopy(values, 0, tempArrayList, 0, index);
+            System.arraycopy(values, index + 1, tempArrayList, index, values.length - index - 1);
+            values = tempArrayList;
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Do not have this index");
+        }
+        return values[index];
+    }
+
+    @Override
+    public void add(T valueOfElement) {
+        T[] tempArrayForValues = (T[]) values;
+        values = (T[]) new Object[tempArrayForValues.length + tempArrayForValues.length >> 1];
+        System.arraycopy(tempArrayForValues, 0, values, 0, tempArrayForValues.length);
+        values[values.length - 1] = valueOfElement;
```

Comment:
if I understand this method will insert value in the last position in the array? //9